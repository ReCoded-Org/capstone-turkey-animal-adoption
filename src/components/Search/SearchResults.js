import React, { useState } from "react";
import { Col, Modal, Button, Card } from "react-bootstrap";
import "./Search.css";
import { BsGeoAlt } from "react-icons/bs";
import { FaPaw } from "react-icons/fa";

const SearchResults = ({ img, id, location, age, breed, gender }) => {
    const [show, setShow] = useState(false);

    const handleToggle = () => setShow(!show);

    return (
        <>
            <Col lg={4} className="mb-5">
                {/* <img src={img} alt="image" className="searchImage"/> */}
                <Card style={{ width: '18rem' }}>
                    <Card.Img variant="top" src={img} />
                    <Card.Body id="modalBody">
                        <Card.Text id="cardText">
                            <div className="firstDiv">
                                <p>{breed}</p>
                                <Button variant="primary" className="adoptButton" onClick={() => handleToggle()}>
                                    <FaPaw className="pawIcon"/>
                                    Adopt
                                </Button>
                            </div>
                            <div className="secondDiv">
                                <p>{gender}, {age}</p>
                                <p className="mt-4"><BsGeoAlt size={25}/> {location}</p>
                                {/* <BsGeoAlt size={25}/> */}
                            </div>
                        </Card.Text>
                    </Card.Body>
                </Card>
            </Col>
            <Modal show={show} onHide={() => handleToggle()} className="modal-open">
        <Modal.Header closeButton id="modalHeader">
          <Modal.Title>
          </Modal.Title>
        </Modal.Header>
        <Modal.Body id="modalBody">
          <p>My Name is</p>
          <p>I am old</p>
          <p>I am a</p>
        </Modal.Body>
        <Modal.Footer id="modalFooter">
          <Button
            className="shadow-none"
            id="closeButton"
            variant="secondary"
            onClick={() => handleToggle()}
          >
            Close
          </Button>
        </Modal.Footer>
      </Modal>
        </>
    );
};

export default SearchResults;