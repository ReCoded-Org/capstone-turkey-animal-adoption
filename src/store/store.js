import { createStore } from "redux";
import user from "./reducers/usersReducers";

const store = createStore(
  user,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
