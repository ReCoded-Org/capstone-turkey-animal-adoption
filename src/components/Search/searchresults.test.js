import React from "react";
import renderer from "react-test-renderer";
import SearchResults from "./SearchResults.js";
import store from "../../store/store";
import { Provider } from "react-redux";

it("renders SearchResults correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <SearchResults />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
