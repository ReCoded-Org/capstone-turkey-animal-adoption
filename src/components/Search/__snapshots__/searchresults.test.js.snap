// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchResults correctly 1`] = `
<div
  className="mb-5 cardCol col-lg-4"
>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
      id="modalBodySearch"
    >
      <div
        id="cardText"
      >
        <div
          className="firstDiv"
        >
          <p />
          <p />
          <button
            className="shadow-none btn btn-primary"
            disabled={false}
            id="adoptButton"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="pawIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 224c-79.41 0-192 122.76-192 200.25 0 34.9 26.81 55.75 71.74 55.75 48.84 0 81.09-25.08 120.26-25.08 39.51 0 71.85 25.08 120.26 25.08 44.93 0 71.74-20.85 71.74-55.75C448 346.76 335.41 224 256 224zm-147.28-12.61c-10.4-34.65-42.44-57.09-71.56-50.13-29.12 6.96-44.29 40.69-33.89 75.34 10.4 34.65 42.44 57.09 71.56 50.13 29.12-6.96 44.29-40.69 33.89-75.34zm84.72-20.78c30.94-8.14 46.42-49.94 34.58-93.36s-46.52-72.01-77.46-63.87-46.42 49.94-34.58 93.36c11.84 43.42 46.53 72.02 77.46 63.87zm281.39-29.34c-29.12-6.96-61.15 15.48-71.56 50.13-10.4 34.65 4.77 68.38 33.89 75.34 29.12 6.96 61.15-15.48 71.56-50.13 10.4-34.65-4.77-68.38-33.89-75.34zm-156.27 29.34c30.94 8.14 65.62-20.45 77.46-63.87 11.84-43.42-3.64-85.21-34.58-93.36s-65.62 20.45-77.46 63.87c-11.84 43.42 3.64 85.22 34.58 93.36z"
              />
            </svg>
            Adopt
          </button>
        </div>
        <div
          className="secondDiv"
        >
          <p>
            , 
          </p>
          <p
            className="mt-4"
          >
            <svg
              fill="currentColor"
              height={25}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width={25}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M8 16s6-5.686 6-10A6 6 0 002 6c0 4.314 6 10 6 10zm0-7a3 3 0 100-6 3 3 0 000 6z"
                fillRule="evenodd"
              />
            </svg>
             
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
